### PR TITLE
Create mbed-studio.md

### PR DESCRIPTION
**Pull request by AnotherButler**
Create initial page for Arm Mbed Studio.

### CODE REVIEW

File: docs/tools/offline/mbed-studio.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+#### Explore or edit the project
+
+Mbed Studio provides support for formatting, syntax highlighting, locating files, comparing files and navigating to lower level code, such as pin definitions for your target hardware. For a more general look at the features provided by Eclipse, consult the [Eclipse](http://help.eclipse.org/neon/index.jsp) online help documentation.
+
+Now, there is a project located in the `C++ Projects` with the name `mbed-os-example-blinky` (or whatever project name you provided earlier). Clicking the triangle next to the project expands its contents and reveals the project’s current files and directories.
+
+Double-click on the `main.cpp` file to open it in Mbed Studio’s editor area. This file includes the code needed to blink an LED. If you haven’t done so already, connect an Mbed Enabled board (such as the FRDM-K64F) to your machine. Mbed Studio detects the board and prompts you with a dialog asking to switch targets. Accept this, and you are ready to run the example.
```
Comment (AnotherButler) [inline docs/tools/offline/mbed-studio.md]: Query: Why have we chosen the FRDM-K64F as our example board for this document?

File: docs/tools/offline/mbed-studio.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+#### Running the project
+
+This section covers the necessary steps of building and running the project, so the program deploys and runs a connected board. Mbed Studio must recognize the board for the instructions in the Run section to work. This documentation uses the FRDM-K64F throughout.
```
Comment (AnotherButler) [inline docs/tools/offline/mbed-studio.md]: Repeated query: Why have we chosen the FRDM-K64F as our example board?

File: docs/tools/offline/mbed-studio.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
+<h2 id="mbed-studio">Arm Mbed Studio</h2>
+
+Arm Mbed Studio is a local development environment for Mbed OS projects. This document helps Mbed OS developers use Mbed Studio to code, run, debug and deploy Mbed OS programs to Mbed Enabled devices. The Mbed Studio project is based on the Eclipse framework and extends Eclipse’s IDE capability to provide a more focused experience for developing with Mbed OS. In particular, it allows users to develop and deploy code offline. It also provides access to existing online resources.
+
+### Current status and downloads
+
+Mbed Studio is currently in closed alpha. You can request access here if you're happy to provide feedback. During this phase, some features are still in progress, and there are frequent updates to the product.
```
Comment (loverdeg-ep) [inline docs/tools/offline/mbed-studio.md]: @AnotherButler  I'd be extremely happy to provide feedback please.

File: docs/tools/offline/mbed-studio.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+##### Simple debugging
+
+To begin debugging, click the bug icon, which launches the default debug configuration.
+
+Alternatively, you can create the debug launch configuration and then begin debugging. To create a debug launch configuration, right-click on a project and select `Debug As... > Debug Configurations` from the context menu. (You can also use the menu bar as shown below.)
+
+Once created, the configuration prepopulates all the readily inferred values. Click `Choose...` next to the project textbox, and select the project you wish to debug if it isn’t already populated. After that, connect a board, and the project is ready to debug. The Mbed Debug launch configuration prepopulates the PyOCD GDB server and the GDB client to locations to the internal Mbed Studio tools. Select the Debug button in the lower right portion of the window to start debugging.
```
Comment (AnotherButler) [inline docs/tools/offline/mbed-studio.md]: Query: What are readily inferred values? Is there another way to write this?

File: docs/tools/offline/mbed-studio.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+Mbed Studio automatically switches into the Mbed Debug perspective when debugging with the Mbed Debug launch configuration. The project also breakes on a default break point. The resulting workspace looks like the image below.
```
Comment (AnotherButler) [inline docs/tools/offline/mbed-studio.md]: Note: There is no image below. Please add the image to the "images" folder at the top level of these docs, and include the AWS link in this .md file.

File: docs/tools/offline/mbed-studio.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+
+##### Advanced debugging
+
+If you want more control over the options regarding debugging, this section is for you. This section describes how to create a PyOCD GDB debug configuration. This configuration allows you to change more settings, such as ports and executable locations for the GDB client and server.
+
+To begin debugging, create the debug launch configuration and set some variables. (You may use your own values if you don’t want to use the default provided here.) To create a debug launch configuration, as before, right-click on a project, and select `Debug As... > Debug Configurations` from the context menu.
+
+A window for the debug configurations appears. Select the `GDB PyOCD Debugging` category, and create a new launch configuration. 
+
+You can click the new icon (a page with a `+` in the upper right corner), or right-click on the category and select `New`. See the screenshot above on the right.
```
Comment (AnotherButler) [inline docs/tools/offline/mbed-studio.md]: Note: There is no image above. Please add the image to the "images" folder at the top level of these docs, and include the AWS link in this .md file.

File: docs/tools/offline/mbed-studio.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+
+Next, configure the binary you want to use for debugging. This is the `.elf` file located within the project directory and inside `BUILD/<target>/GCC_ARM`. (For this example, it is `BUILD/K64F/GCC_ARM`.) Currently, this is configured manually, so switching to a different target to run the same project requires you to update this field in the debug configuration. See the below left screenshot.
```
Comment (AnotherButler) [inline docs/tools/offline/mbed-studio.md]: Note: There is no image below left. Please add the image to the "images" folder at the top level of these docs, and include the AWS link in this .md file.

File: docs/tools/offline/mbed-studio.md (hunk 3 above)
Comment (AnotherButler) [reply docs/tools/offline/mbed-studio.md]: @JoeTheGuitarist Is this something you can help with?

File: docs/tools/offline/mbed-studio.md [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+`\eclipse-ws\mbed-studio-ws\tools\python\Scripts\pyocd-gdbserver.exe`
+`\eclipse-ws\mbed-studio-ws\tools\gcc-arm\bin\gcc-arm-none-eabi-gdb.exe`
+
+When you have configured everything, select `Apply` in the lower right side of the panel, and the click the blue Debug button just below that.
+
+### Help
+
+This section outlines known issues and provides answers to common questions.
+
+#### Glossary
```
Comment (AnotherButler) [inline docs/tools/offline/mbed-studio.md]: Query: Would you prefer we move these terms to the overall Mbed OS glossary? If not, shall we duplicate them there?

File: docs/tools/offline/mbed-studio.md (hunk 3 above)
Comment (JoeTheGuitarist) [reply docs/tools/offline/mbed-studio.md]: @loverdeg-ep we'll be putting up the signup form soon, so assuming you have an Mbed account, I'd be more than happy to give you access. Thanks!

File: docs/tools/offline/mbed-studio.md (hunk 1 above)
Comment (JoeTheGuitarist) [reply docs/tools/offline/mbed-studio.md]: Good point, there's no reason in particular, I think that's just what Matt was using as a reference, but we can make it board-agnostic.

File: docs/tools/offline/mbed-studio.md (hunk 4 above)
Comment (JoeTheGuitarist) [reply docs/tools/offline/mbed-studio.md]: No idea....! That sentence can be deleted.

File: docs/tools/offline/mbed-studio.md (hunk 5 above)
Comment (JoeTheGuitarist) [reply docs/tools/offline/mbed-studio.md]: Ah yes, the original docs have lots of screenshots in them, which I removed for ease of updating this in future. I can either add the images back in, or delete any mention of them.

File: docs/tools/offline/mbed-studio.md (hunk 8 above)
Comment (JoeTheGuitarist) [reply docs/tools/offline/mbed-studio.md]: Fine to move them - as long as we link to them.

**Pull request merged**